lib/fog/vsphere/requests/compute/modify_vm_interface.rb

```diff
@@ -19,11 +19,22 @@ def destroy_vm_interface(vmid, options = {})
         def update_vm_interface(vmid, options = {})
           raise ArgumentError, "instance id is a required parameter" unless vmid
 
-          interface = get_interface_from_options(vmid, options.merge(:server_id => vmid))
-          vm_reconfig_hardware('instance_uuid' => vmid, 'hardware_spec' => {'deviceChange'=>[create_interface(interface, interface.key, :edit)]})
+          interface = get_interface_from_options(vmid, options)
+          raw_interface = get_raw_interface(vmid, key: interface.key)
+          if options[:network]
+              interface.network = options[:network]
```

Comment:
Oops - sorry! My editor is set to 4-spaces by default and I keep forgetting to set it to 2 for these files.